_config/email.yml

```diff
@@ -1,12 +1,53 @@
 ---
-Name: emailconfig
+Name: mailer
 ---
 SilverStripe\Core\Injector\Injector:
-  Swift_Transport: Swift_MailTransport
-  Swift_Mailer:
-    constructor:
-      - '%$Swift_Transport'
-  SilverStripe\Control\Email\Mailer:
-    class: SilverStripe\Control\Email\SwiftMailer
-    properties:
-      SwiftMailer: '%$Swift_Mailer'
+  Symfony\Component\Mailer\MailerInterface:
+    class: Symfony\Component\Mailer\Mailer
+    constructor:
+      0: '%$Symfony\Component\Mailer\Transport\TransportInterface'
+      1: '%$Symfony\Component\Messenger\MessageBusInterface.mailer'
+  Symfony\Component\Messenger\MessageBusInterface.mailer:
+    class: Symfony\Component\Messenger\MessageBus
+    constructor:
+      0:
+        - '%$Symfony\Component\Messenger\Middleware\MiddlewareInterface.mailer'
+  Symfony\Component\Messenger\Middleware\MiddlewareInterface.mailer:
+    class: Symfony\Component\Messenger\Middleware\SendMessageMiddleware
+    constructor:
+      0: '%$Symfony\Component\Messenger\Transport\Sender\SendersLocatorInterface.mailer'
+      1: '%$Symfony\Component\EventDispatcher\EventDispatcherInterface.mailer'
+  Symfony\Component\Messenger\Transport\Sender\SendersLocatorInterface.mailer:
+    class: Symfony\Component\Messenger\Transport\Sender\SendersLocator
+    constructor:
+      0:
+        '*':
+          - 'Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer'
+      1: '%$Symfony\Component\DependencyInjection\ContainerInterface.mailer'
+  Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer:
+    class: SilverStripe\Control\Email\EmailSender
+  Symfony\Component\DependencyInjection\ContainerInterface.mailer:
+    class: Symfony\Component\DependencyInjection\Container
+    calls:
+      - ['set', ['Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer', '%$Symfony\Component\Messenger\Transport\Sender\SenderInterface.mailer']]
+  Symfony\Component\EventDispatcher\EventDispatcherInterface.mailer:
+    class: Symfony\Component\EventDispatcher\EventDispatcher
+---
+Name: mailer-dsn-default-config
+---
+SilverStripe\Core\Injector\Injector:
+  Symfony\Component\Mailer\Transport\TransportInterface:
+    factory: Symfony\Component\Mailer\Transport
+    factory_method: fromDsn
+    constructor:
+      dsn: 'sendmail://default'
+---
+Name: mailer-dsn-env
+After: mailer-dsn-default-config
+Only:
+  envvarset: MAILER_DSN
+---
+SilverStripe\Core\Injector\Injector:
+  Symfony\Component\Mailer\Transport\TransportInterface:
+    constructor:
+      dsn: '`MAILER_DSN`'
```

Comment:
I agree with your assessment in https://github.com/silverstripe/developer-docs/pull/92#discussion_r984055645 - this should override all other config. Please remove this from yml and instead have the env variable as an override in the PHP logic.

Reply:
Have updated - as noted in https://github.com/silverstripe/developer-docs/pull/92#discussion_r988384245 I've retained the yml config instead of doing in PHP